users.go

```diff
@@ -169,22 +171,51 @@ func (api *Client) GetUserInfoContext(ctx context.Context, user string) (*User,
 	return &response.User, nil
 }
 
-// GetUsers returns the list of users (with their detailed information)
+// GetUsers returns the list of users (with their detailed information). This
+// call can take a long time for large teams, due to rate limits on `users.list`.
 func (api *Client) GetUsers() ([]User, error) {
 	return api.GetUsersContext(context.Background())
 }
 
-// GetUsersContext returns the list of users (with their detailed information) with a custom context
-func (api *Client) GetUsersContext(ctx context.Context) ([]User, error) {
+// GetUsersContext returns the list of users (with their detailed information)
+// with a custom context. This call can take a long time for large teams, due to
+// rate limits on `users.list`.
+func (api *Client) GetUsersContext(ctx context.Context) (users []User, err error) {
```

Comment:
since You're adding a new method, return the pagination id and accept it as a parameter. don't want to restrict users of the library from being constrainted to storing all the users in memory.